GTG/core/datastore2.py

```diff
@@ -0,0 +1,384 @@
+        backup_file = f"{filename}.bak.{i}" if i else filename
+
+        return os.path.join(dirname, 'backup', backup_file)
+
+
+    def write_backups(self, path: str) -> None:
+        current_backup = self.BACKUPS
+        backup_name = self.get_backup_name(filepath)
+        backup_dir = os.path.dirname(backup_name)
+
+        # Make sure backup dir exists
+        try:
+            os.makedirs(backup_dir, exist_ok=True)
+
+        except IOError:
+            log.error('Backup dir %r cannot be created!', backup_dir)
+            return
+
+        # Cycle backups
+        while current_backup > 0:
+            older = f"{backup_name}.bak.{current_backup}"
+            newer = f"{backup_name}.bak.{current_backup - 1}"
+
+            if os.path.exists(newer):
+                shutil.move(newer, older)
+
+            current_backup -= 1
+
+        # bak.0 is always a fresh copy of the closed file
+        # so that it's not touched in case of not opening next time
+        bak_0 = f"{backup_name}.bak.0"
+        shutil.copy(filepath, bak_0)
+
+        # Add daily backup
+        today = datetime.today().strftime('%Y-%m-%d')
+        daily_backup = f'{backup_name}.{today}.bak'
+
+        if not os.path.exists(daily_backup):
+            shutil.copy(filepath, daily_backup)
+
+
+    def find_and_load_file(self, path: str) -> str:
+        """Find an XML file to open
+
+        If file could not be opened, try:
+            - file__
+            - file.bak.0
+            - file.bak.1
+            - .... until BACKUP_NBR
+
+        If file doesn't exist, create a new file."""
+
+        files = [
+            xml_path,            # Main file
+            xml_path + '__',     # Temp file
+        ]
+
+        # Add backup files
+        files += [self.get_backup_name(xml_path, i)
+                  for i in range(self.BACKUPS)]
+
+        backup_info = None
+
+        for index, filepath in enumerate(files):
+            try:
+                log.debug('Opening file %s', filepath)
+                self.load_file(filepath)
+
+                timestamp = os.path.getmtime(filepath)
+                mtime = datetime.fromtimestamp(timestamp).strftime('%Y-%m-%d')
+
+                # This was a backup. We should inform the user
+                if index > 0:
+                    backup_info = {
+                        'name': filepath,
+                        'time': mtime
+                    }
+
+                # We could open a file, let's stop this loop
+                break
+
+            except FileNotFoundError:
+                log.debug('File not found: %r. Trying next.', filepath)
+                continue
+
+            except PermissionError:
+                log.debug('Not allowed to open: %r. Trying next.', filepath)
+                continue
+
+            except etree.XMLSyntaxError as error:
+                log.debug('Syntax error in %r. %r. Trying next.',
+                          filepath, error)
+                continue
```

Comment:
Maybe save that as well and maybe report to the user? Like another dialog box saying that it found files but couldn't open them (and thus loaded the defaulted one)